### ISSUE TITLE
bug: `COPY INTO` CPU load takes a long time to rise

### ISSUE THREAD

**Issue by yufan022** [datafuselabs/databend#8574]
### Search before asking

- [X] I had searched in the [issues](https://github.com/datafuselabs/databend/issues) and found no similar issues.

### Version

v0.8.89-nightly

### What's Wrong?

```
COPY INTO import FROM 's3://xx/xx/100000-100/' credentials=(aws_key_id='xx' aws_secret_key='xx') pattern ='.*[.]gz' file_format = (type = 'tsv' compression
= GZIP) force=true;
Query OK, 0 rows affected (2 min 22.85 sec)
```
The s3 path /100000-100/* only has 100 files.

At the beginning of the command execution, I observed a very low cpu load:
![image](https://user-images.githubusercontent.com/30121694/199174706-6b8efd45-cc9e-46f6-ab17-50696d240b9f.png)

It took about 60 seconds for the CPU load to rise:
![image](https://user-images.githubusercontent.com/30121694/199174745-ac830142-3161-4672-a7c1-80e962440ad7.png)

i checked info logs during the period of low CPU load.
![image](https://user-images.githubusercontent.com/30121694/199175951-91db5ad3-e060-484d-aa0e-63e8256a5b11.png)
got alot of this
```
2022-11-01T06:47:01.463339Z  INFO opendal::services::s3::backend: backend build started: Builder
2022-11-01T06:47:01.463371Z  INFO opendal::services::s3::backend: backend use root
2022-11-01T06:47:02.067443Z  INFO opendal::services::s3::backend: backend build finished: Builder
2022-11-01T06:47:01.463339Z  INFO opendal::services::s3::backend: backend build started: Builder
2022-11-01T06:47:01.463371Z  INFO opendal::services::s3::backend: backend use root
2022-11-01T06:47:02.067443Z  INFO opendal::services::s3::backend: backend build finished: Builder
2022-11-01T06:47:01.463339Z  INFO opendal::services::s3::backend: backend build started: Builder
2022-11-01T06:47:01.463371Z  INFO opendal::services::s3::backend: backend use root
2022-11-01T06:47:02.067443Z  INFO opendal::services::s3::backend: backend build finished: Builder
...
```

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

**Comment by BohuTANG** [comment]
Ping @Xuanwo  for the first-glance troubleshooting. Many `opendal::services::s3::backend` logs is by design?

**Comment by Xuanwo** [comment]
> Ping @Xuanwo for the first-glance troubleshooting. Many `opendal::services::s3::backend` logs is by design?

Yes.

I'm guessing they are plenty of `read` (which is in debug) logs here.

**Comment by yufan022** [comment]
> > Ping @Xuanwo for the first-glance troubleshooting. Many `opendal::services::s3::backend` logs is by design?
> 
> Yes.
> 
> I'm guessing they are plenty of `read` (which is in debug) logs here. However, it's still worth to reuse the same operator to reduce a bit of cost.

Is it concurrent? This log will last for one minute, during which the network traffic is not large.

**Comment by Xuanwo** [comment]
> Is it concurrent? The log will last for one minute, the speed is not fast, and there is not much network traffic during this period.

Most of the time is consumed on parsing CSV content. We will improve it at https://github.com/datafuselabs/databend/issues/8486.

**Comment by BohuTANG** [comment]
I can not reproduce this case with TSV (10000 rows for each file, 100 files, ~43MB each file), data is generated by bendlog.

My `COPY` script:
```
n=0
while true; do
        mysql -uroot -h127.0.0.1 -P3307 -e "COPY into bendlog.log from @log pattern='.*[.]csv' FILE_FORMAT = (TYPE = 'TSV' SKIP_HEADER=1) force=true"
        echo $n
        n=$((n+1))
        sleep 0.1s
done
```

From the `dstat` command, we can see it always doing work:
<img width="610" alt="image" src="https://user-images.githubusercontent.com/172204/199187527-c5bcf198-4ad2-4744-b867-b97868d3ecf2.png">

**Comment by Xuanwo** [comment]
I found the root cause of this issue.

https://github.com/datafuselabs/databend/blob/2dfd578d234fe4ea1f65c432b60bca37ec8b4095/src/query/service/src/interpreters/interpreter_copy_v2.rs#L101-L181

This call requires all files to be `stat` which is not needed. I will improve this.

**Comment by BohuTANG** [comment]
I will try to fix in #8586, cc @Xuanwo
